### DISCUSSION TITLE
Is there some prop/method to "watch" for the open state?

### DISCUSSION THREAD

**Discussion by RSchneider94** [gorhom/react-native-bottom-sheet#1616] [Q&A] [answered]
Hey folks,

I'm currently using another sheet library in my current application, but I'm thinking about moving to this one because my current lib lacks in a feature that would really help me (I have also posted on the lib GitHub about it and asked if they want some contribution, but still waiting for an answer). Basically is the ability to have a `isOpen` or some callback like `onSheetClose`.

This is useful, because for example in my current scenario here: I have a component (which is basically a FlatList that is rendered inside of the Sheet), but this list does receive a prop from the parent component that I have where the Sheet is being called, you can imagine it like an ID which I'll use to do some fetches/edits inside the sheet. Now the problem is, if I have made some changes inside of the list and closed the sheet, the ID will be the same, because the `useState` responsible for the ID still has the same value, because I have no way to tell that the sheet was closed in order to clean it. If I have some onSheetClose method, I could then clear this ID whenever the sheet gets closed.

I took a quick look at the documentation and I'm not sure if this library would fill this gap, I have seem that there's the `index` property where it controls the open/closed state, but I'm not sure if this works also dynamically, like being controlled with `useState` e.g. Have some of you done something like this with this library before?

Thanks!

**Comment by ManuF17** [comment, marked as answer]
@RSchneider94 the BottomSheet component have a onChange prop and you get an index as param in that function, i think 0 is Open state and -1 is closed state, hope this helps to you, maybe before you can go and move out you're current sheet you can make a POC using this library and check what i said here 😅

**Reply by RSchneider94** [reply]
Nice! I think this would do the job for sure. Meanwhile the library I’m currently using accepted my suggestion and is going to implement it soon, so I’m staying with the current one by now haha.

**Comment by RSchneider94** [comment]
> @RSchneider94 the BottomSheet component have a onChange prop and you get an index as param in that function, i think 0 is Open state and -1 is closed state, hope this helps to you, maybe before you can go and move out you're current sheet you can make a POC using this library and check what i said here 😅
